### PR TITLE
new kata: numberGame + tests

### PR DESCRIPTION
**Pull request by AnnaSanna**
Add new kata file NumberGame + test + editing package.json

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,8 +5,8 @@
   "description": "",
   "main": "index.js",
   "scripts": {
-    "test1": "mocha",
-    "test": "c8 --all mocha"
+    "test": "c8 --all mocha",
```
Comment (ivanAndCode) [inline package.json:8]: Good tidy up!

File: numberGame.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+export const errorMessageNoGuessesLeft = 'No guesses left';
+export class Guesser {
+    constructor(number, lives) {
+        this.number = number;
+        this.lives = lives;
+    }
+
+    guess(number) {
+        if (this.lives <= 0) throw new Error(errorMessageNoGuessesLeft);
```
Comment (ivanAndCode) [inline numberGame.js:9]: similar logic should live in a constructor to not allow creating instance with zero/negative lives

File: package.json [hunk 3]
```diff
@@ -5,8 +5,8 @@
...
+    "testNumberGame": "c8 --include numberGame.js mocha test/numberGame.test.js"
```
Comment (ivanAndCode) [inline package.json]: (no idea myself) do you need the include? would it work without it?

File: test/numberGame.test.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
+import { equal } from 'assert';
+import * as numberGame from '../numberGame.js';
+
+it('User guessed the number on the first try', () => {
+    const player = new numberGame.Guesser(12, 1);
+    equal(player.guess(12), true);
+})
+
+it('User guessed the number on the last  try', () => {
```
Comment (ivanAndCode) [inline test/numberGame.test.js]: [nitpick] typo with double space

File: test/numberGame.test.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    const player = new numberGame.Guesser(5, 2);
+    equal(player.guess(12), false);
+    equal(player.guess(5), true);
+})
+
+describe("Should throw an Error", () => {
+    it('Should throw an Error when there is negative number of guesses', () => {
```
Comment (ivanAndCode) [inline test/numberGame.test.js]: test name duplicating the describe

File: test/numberGame.test.js [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+        const player = new numberGame.Guesser(5, -1);
+        try {
+            player.guess(5);
+            equal('Fail', 'For some reason the guess method did not throw an Error');
+        } catch (error) {
+            equal(error.message, numberGame.errorMessageNoGuessesLeft)
+        }
+    })
+
+    it('Should throw an Error when User did not guess the number', () => {
```
Comment (ivanAndCode) [inline test/numberGame.test.js]: [nitpick] modify test name to mention "... with an allowed guess amount" or similar

**Review by ivanAndCode — APPROVED** [review]
Looks good. Left a few notes to tidy up.

File: numberGame.js (hunk 2 above)
Comment (ivanAndCode) [reply numberGame.js:9]: update: should is the wrong word - COULD

Comment (AnnaSanna) [reply numberGame.js:9]: I've decided not to do that

File: test/numberGame.test.js (hunk 4 above)
Comment (AnnaSanna) [reply test/numberGame.test.js]: fixed

File: package.json (hunk 3 above)
Comment (AnnaSanna) [reply package.json]: fixed

File: test/numberGame.test.js (hunk 6 above)
Comment (AnnaSanna) [reply test/numberGame.test.js]: fixed

**Review by ivanAndCode — APPROVED** [review]
LGTM

**Pull request merged**